src/Chainweb/CutDB.hs

```diff
@@ -365,16 +381,22 @@ startCutDb config logfun headerStore payloadStore cutHashesStore = mask_ $ do
     queue <- newEmptyPQueue
     cutAsync <- asyncWithUnmask $ \u -> u $ processor queue cutVar
     logg Info "CutDB started"
-    return $! CutDb
-        { _cutDbCut = cutVar
-        , _cutDbQueue = queue
-        , _cutDbAsync = cutAsync
-        , _cutDbLogFunction = logfun
-        , _cutDbHeaderStore = headerStore
-        , _cutDbPayloadStore = payloadStore
-        , _cutDbQueueSize = _cutDbParamsBufferSize config
-        , _cutDbStore = cutHashesStore
-        }
+    let
+        !db = CutDb
+            { _cutDbCut = cutVar
+            , _cutDbQueue = queue
+            , _cutDbAsync = cutAsync
+            , _cutDbLogFunction = logfun
+            , _cutDbHeaderStore = headerStore
+            , _cutDbPayloadStore = payloadStore
+            , _cutDbQueueSize = _cutDbParamsBufferSize config
+            , _cutDbStore = cutHashesStore
+            }
+    pruneCuts db
+    -- we compact the entire cut hashes table on startup in case we just 
+    -- pruned a lot of cut hashes
+    compactRangeRocksDb (_getRocksDbCas cutHashesStore) (Nothing, Nothing)
```

Comment:
is this synchronous? How long can it take?

Reply:
It can take quite a while, but this is fixed in #1342. In fact I will close this PR and address your feedback in that PR.